Docs/developer_guide/python_faq.md

```diff
@@ -2,6 +2,85 @@
 
 Frequently asked questions about how to write Python scripts for Slicer.
 
+## What is the Slicer Python Environment?
+
+You can consider each Slicer installation as a virtual Python environment - the same way
+as you create virtual environments using python or conda.
+
+## What is the PythonSlicer executable?
+
+`PythonSlicer` is an executable provided in the `bin` directory of the Slicer installation.
+It’s the same version of Python available through the Slicer application, but it acts like
+a normal `python` (or `python3`) executable in terms of running scripts and can be used t
+ access all the packages installed in Slicer.
+
+This means you can use it in place of regular python to make use of the Slicer environment
+for batch processing and command-line operations.
```

Comment:
Some examples would be great here.

Reply:
Ditto. Here is an issue tracking this:
* https://github.com/Slicer/Slicer/issues/7015


Since depending on the needed capabilities batch processing may have to be implemented through the ``Python Console by leveraging `--python-script` and `--no-main-windows`, I expanded the `What is the Python Console` section with the following:


```
:::{hint}
Running scripts (or code) using the command-line option `--python-script` (or `python-code`)
is equivalent to running code in the Python Console.

Combined with the use of `--no-main-window`, this is useful for implementing batch processing
pipelines leveraging capabilities only available in the context of a running application.
For example, this applies to the [Segment Editor effects](/user_guide/modules/segmenteditor.html#effects).
:::
```

Reply:
I think I found a nice compromise in the context of the API documentation:

![image](https://github.com/Slicer/Slicer/assets/219043/e16118e1-59d6-4e09-af8d-99294383a936)